modules/programs/powerline-go.nix

```diff
@@ -0,0 +1,121 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.powerline-go;
+
+  # Convert an option value to a string to be passed as argument to
+  # powerline-go:
+  valueToString = value:
+    if builtins.isList value then
+      builtins.concatStringsSep "," (builtins.map valueToString value)
+    else if builtins.isAttrs value then
+      valueToString
+      (mapAttrsToList (key: val: "${valueToString key}=${valueToString val}")
+        value)
+    else
+      builtins.toString value;
+
+  modulesArgument = optionalString (cfg.modules != null)
+    "-modules ${valueToString cfg.modules}";
+
+  newlineArgument = optionalString cfg.newline "-newline";
+
+  pathAliasesArgument = optionalString (cfg.pathAliases != null)
+    "-path-aliases ${valueToString cfg.pathAliases}";
+
+  otherSettingPairArgument = name: value:
+    if value == true then "-${name}" else "-${name} ${valueToString value}";
+
+  otherSettingsArgument = optionalString (cfg.settings != { })
+    (concatStringsSep " "
+      (mapAttrsToList otherSettingPairArgument cfg.settings));
+
+  commandLineArguments = ''
+    ${modulesArgument} ${newlineArgument} ${pathAliasesArgument} ${otherSettingsArgument}
+  '';
+
+in {
+  meta.maintainers = [ maintainers.DamienCassou ];
+
+  options = {
+    programs.powerline-go = {
+      enable = mkEnableOption "Powerline-go";
+
+      modules = mkOption {
+        default = null;
+        type = types.nullOr (types.listOf types.str);
+        description = ''
+
+          List of module names to load. The list of all available
+          modules as well as the choice of default ones are at <link
+          xlink:href="https://github.com/justjanne/powerline-go">https://github.com/justjanne/powerline-go</link>.
+
+        '';
+        example = [ "host" "ssh" "cwd" "gitlite" "jobs" "exit" ];
+      };
+
+      newline = mkOption {
+        default = false;
+        type = types.bool;
+        description =
+          "Set to true if the prompt should be on a line of its own.";
+        example = true;
+      };
+
+      pathAliases = mkOption {
+        default = null;
+        type = types.nullOr (types.attrsOf types.str);
+        description = ''
+          Pairs of full-path and corresponding desired short name. You
+          may use '~' to represent your home directory but you should
+          protect it to avoid shell substitution.
+        '';
+        example = { "\\~/projects/home-manager" = "prj:home-manager"; };
+      };
+
+      settings = mkOption {
+        default = { };
+        type = with types;
+          attrsOf (either bool (either int (either str (listOf str))));
```

Comment:
I've just learned about `types.oneOf`. Should I use that instead?

Reply:
I would guess so. makes it easier to read

Reply:
done